packages/ButtonGroup/src/ButtonGroup.js

```diff
@@ -0,0 +1,165 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import ButtonGroupContext from "./ButtonGroupContext";
+import Item from "./components/Item";
+import * as sc from "./ButtonGroup.styles";
+
+const propTypes = {
+  /** The toggle buttons in the group. */
+  children: PropTypes.node,
+
+  /** To show the icons used for selected / not selected. */
+  hasIcons: PropTypes.bool,
+
+  /** If the entire button group is disabled. */
+  isDisabled: PropTypes.bool,
+
+  /** If the width of the button group should stretch to fit its parent container (100%). */
+  isFullWidth: PropTypes.bool,
+
+  /** If multiple simultaneous selections are allowed. */
+  isMulti: PropTypes.bool,
+
+  /** If it will be rendered as a button element. If false, a span will be rendered via an accessible RawButton. */
+  isSemantic: PropTypes.bool,
+
+  /** Callback to be executed when any button item is clicked or activated by keyboard. It will return an array of the selected items' "value" properties. */
+  onChange: PropTypes.func,
+
+  /** Size of the buttons (height, font size, etc). */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  children: null,
+  hasIcons: false,
+  isDisabled: false,
+  isFullWidth: false,
+  isMulti: false,
+  isSemantic: true,
+  onChange: () => {},
+  size: ShirtSizes.MEDIUM,
+};
+
+const keyTypes = {
+  PREV: "ArrowLeft",
+  NEXT: "ArrowRight",
+  FIRST: "Home",
+  LAST: "End",
+};
+
+function ButtonGroup(props) {
+  const { children, hasIcons, isDisabled, isMulti, isSemantic, onChange, size } = props;
+
+  const [selectedItems, setSelectedItems] = React.useState([]);
+  const [currentFocusIndex, setFocusIndex] = React.useState(null);
+  const [currentFocusValue, setFocusValue] = React.useState(null);
+
+  const groupRef = React.createRef();
+
+  function isItemDisabled(item) {
+    return item.getAttribute("aria-disabled") === "true" || item.hasAttribute("disabled");
+  }
+
+  function getButtonRefs() {
+    return Array.from(groupRef.current.querySelectorAll("[data-pka-anchor='button-group.button']"));
+  }
+
+  function getEnabledIndexes() {
+    const buttonRefs = getButtonRefs();
+    return buttonRefs.map((item, index) => (isItemDisabled(item) ? null : index)).filter(index => index !== null);
+  }
+
+  React.useLayoutEffect(() => {
```

Comment:
Was there problems using just `React.useEffect` here?

Reply:
Just that it would render without the selection for a split second. Same reason Alex recommended `useLayoutEffect` here:
https://github.com/acl-services/paprika/pull/469#discussion_r422331013